glusterd2/commands/volumes/volume-reset.go

```diff
@@ -89,17 +103,36 @@ func volumeResetHandler(w http.ResponseWriter, r *http.Request) {
 		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
 		return
 	}
-
+	//save volume information for transaction failure scenario
+	if err := txn.Ctx.Set("oldvolinfo", oldvolinfo); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+	opt := make(map[string]string)
```

Comment:
I"m sorry @Madhu-1. I didn't realise that - one request was of map type and the other one was a slice. hence this doesn't look as clean. Duplicating `isActionStepRequired` might be cleaner. I'll let this one go though.

Reply:
if necessary will do clean up in upcoming PR.